### PR TITLE
Allow skipping docker pull during `get` step [#100994844]

### PR DESCRIPTION
**Pull request by monkeyherder**

### CODE REVIEW

File: assets/in [hunk 1]
```diff
@@ -37,22 +37,40 @@ fi
...
+skip_download="$(jq -r '.params.skip_download // false' < $payload)"
 
 mkdir -p $destination
 
-docker save -o ${destination}/image "$image"
-
-if [ "$rootfs" = "true" ]; then
-  docker run --cidfile=container.cid "$image" echo container created
-  docker export $(cat container.cid) > ${destination}/rootfs.tar
+if ! $skip_download
```
Comment (xoebus) [inline assets/in]: This should be of the form:

```
if [ "$skip_download" = "true" ]; then
```

to avoid subtle `bash` behavior.

**Comment by monkeyherder** [comment]
@xoebus Cleaned up to avoid `bash` subtleties.

File: assets/in [hunk 2]
```diff
@@ -37,22 +37,39 @@ fi
...
+  if [ "$rootfs" = "true" ]; then
+    docker run --cidfile=container.cid "$image" echo container created
+    docker export $(cat container.cid) > ${destination}/rootfs.tar
+  fi
+else
+  if private_registry "${repository}"
+  then
+    registry=$(extract_registry $repository)
+    repository=$(extract_repository $repository)
+    json_selector="."
```
Comment (vito) [inline assets/in]: is docker really inconsistent within its own "v1" API?

:'(

**Comment by monkeyherder** [comment]
Yeah, docker really is incompatible with its own "v1" API. See https://github.com/docker/docker-registry/issues/517#issuecomment-52786954 . While the discussion is about the `tags` endpoint, rather than `tags/<version>`, we saw the same sort of inconsistent behavior.

**Pull request merged**